README.md

```diff
@@ -306,6 +306,510 @@ val byteToHex = FunSpec.builder("byteToHex")
+            .returns(String::class)
+            .build())
+        .build())
+    .addEnumConstant("SCISSORS", TypeSpec.anonymousClassBuilder()
+        .addSuperclassConstructorParameter("%S", "peace")
+        .build())
+    .addEnumConstant("PAPER", TypeSpec.anonymousClassBuilder()
+        .addSuperclassConstructorParameter("%S", "flat")
+        .build())
+    .addProperty(PropertySpec.builder("handsign", String::class, KModifier.PRIVATE)
+        .initializer("handsign")
+        .build())
+    .build()
+```
+
+Which generates this:
+
+```kotlin
+enum class Roshambo(private val handsign: kotlin.String) {
+    ROCK("fist") {
+        override fun toString(): kotlin.String = "avalanche!"
+    },
+
+    SCISSORS("peace"),
+
+    PAPER("flat");
+}
+```
+
+### Anonymous Inner Classes
+
+In the enum code, we used `TypeSpec.anonymousClassBuilder()`. Anonymous inner classes can also be 
+used in code blocks. They are values that can be referenced with `%L`:
+
+```kotlin
+val comparator = TypeSpec.anonymousClassBuilder()
+    .addSuperinterface(Comparator::class.parameterizedBy(String::class))
+    .addFunction(FunSpec.builder("compare")
+        .addModifiers(KModifier.OVERRIDE)
+        .addParameter("a", String::class)
+        .addParameter("b", String::class)
+        .returns(Int::class)
+        .addStatement("return %N.length - %N.length", "a", "b")
+        .build())
+    .build()
+
+val helloWorld = TypeSpec.classBuilder("HelloWorld")
+    .addFunction(FunSpec.builder("sortByLength")
+        .addParameter("strings", List::class.parameterizedBy(String::class))
+        .addStatement("%N.sortedWith(%L)", "strings", comparator)
+        .build())
+    .build()
+```
+
+This generates a method that contains a class that contains a method:
+
+```kotlin
+class HelloWorld {
+    fun sortByLength(strings: kotlin.collections.List<kotlin.String>) {
+        strings.sortedWith(object : java.util.Comparator<kotlin.String> {
+            override fun compare(a: kotlin.String, b: kotlin.String): kotlin.Int = a.length - b.length
+        })
+    }
+}
+```
+
+One particularly tricky part of defining anonymous inner classes is the arguments to the superclass
+constructor. To pass them use `TypeSpec.Builder`'s `addSuperclassConstructorParameter()` method. 
+
+### Annotations
+
+Simple annotations are easy:
+
+```kotlin
+val test = FunSpec.builder("test string equality")
+    .addAnnotation(Test::class)
+    .addStatement("assertThat(%1S).isEqualTo(%1S)", "foo")
+    .build()
+```
+
+Which generates this function with an `@Test` annotation:
+
+```kotlin
+@kotlin.Test
+fun `test string equality`() {
+    assertThat("foo").isEqualTo("foo")
+}
+```
+
+Use `AnnotationSpec.builder()` to set properties on annotations:
+
+```kotlin
+val logRecord = FunSpec.builder("recordEvent")
+    .addModifiers(KModifier.ABSTRACT)
+    .addAnnotation(AnnotationSpec.builder(Headers::class)
+        .addMember("accept = %S", "application/json; charset=utf-8")
```

Comment:
should we change the API so that "accept" is one argument, and its value is another?

Reply:
That's what we had initially and decided to simplify: 455c3a67a8cba64ed7ad1f3296384e1ff593f187